### ISSUE TITLE
Use different display labels for data.

### ISSUE THREAD

**Issue by SamMousa** [nicolaskruchten/pivottable#574]
I'd love to be able to specify labels for attributes and or values.
Consider a situation where my data looks like this:
````
[{
    text: "test1123",
    c1: 53
    c2: 12
}...]
````
Now I'd like to use this data but when drawing labels I might want to use different labels like `title`, `longitude`, `latitude`.
Of course I could  modify the data, but in a dynamic setting doing this just before rendering is more efficient (if I have 100.000 rows I still only draw the headers once)...

**Comment by nicolaskruchten** [comment]
You could use the `derivedAttributes` system for this...

**Comment by SamMousa** [comment]
Yes I know, my point is that this is inefficient since it'll have to call the function for each data point instead of just for rendering the labels.
It'll increase memory usage and cpu usage. Wouldn't it be better to have some optional callbacks that are only called when rendering labels / values?

**Comment by SamMousa** [comment]
Just did a little test, a simple derived attributes on a set with ~ 1000 records. The function gets called on every record each time something changes in the table; that's not ideal.

**Comment by SamMousa** [comment]
The current implementation calls the function for derived attributes a lot; this means that for a dataset of ~1000 records I see ~33.000 calls after doing ~5 drags / drop operations. 
This probably is because `pivotUI` calls `pivot` with new settings each time, and each time the derivedAttributes are recalculated.

**Comment by nicolaskruchten** [comment]
Agreed that it's inefficient in terms of method calls. Is this causing an actual perceptual problem?

What sort of solution would you have in mind? Is remapping your input before passing it in totally unacceptable? I.e. http://stackoverflow.com/questions/8747561/is-there-any-way-to-rename-js-object-keys-using-underscore-js

**Comment by SamMousa** [comment]
Not totally unacceptable, I'm exploring it as one solution. As well as the
derived attribhtes option.

Currently the pivotData object does some caching and it calls the derived
attributes once for each record. This is a greedy approach; this could be
changed to be lazy combined with memoization.

Second pivotUI should pass the data in a way that does not require the
derived attributes to be reloaded. (Looking at the source the only option
used in the PivotData constructor are the derived attributes and filter
keys).

In conclusion I'm trying to move to a good solution but prefer to spend my
time coding on something useful like extending this library vs coding a
workaround :)

**Comment by nicolaskruchten** [comment]
If you're passing in an array-of-arrays then this type of renaming is trivial: just overwrite the header array.

In terms of modifying the library itself, I'm increasingly resistant to adding new parameters to an already-long list, especially for situations where preprocessing the data before passing it in is fairly trivial...

**Comment by SamMousa** [comment]
Interesting; currently I'm passing in an array of object (which is my source data). The data is dynamically filtered (before it enters pivot table).
I could of course also add these extra columns as getters on the objects and hide the base attributes in pivottables.

**Comment by nicolaskruchten** [comment]
I'm assuming that the issue you raised in #575 is related to this one...

Check out the new example at http://nicolas.kruchten.com/pivottable/examples/mps_rename.html for an efficient way to rename things on the fly using a function to map over your input rather than a raw array.

**Comment by nicolaskruchten** [comment]
See also #492

**Comment by SamMousa** [comment]
I'll look into that; still doing it that way will require creating new objects which is avoidable.

I'll study the source, to see if that would work for my case.

**Comment by nicolaskruchten** [comment]
I'm going to close this as a duplicate of #492 ... please feel free to follow up with any comments there :)
